ports/atmel-samd/Makefile

```diff
@@ -322,6 +322,14 @@ SRC_C += \
 	reset.c \
 	timer_handler.c \
 
+ifeq ($(CIRCUITPY_PWMIO),1)
+SRC_C += shared_timers.c
+endif
+
+ifeq ($(CIRCUITPY_AUDIOIO),1)
+SRC_C += shared_timers.c
+endif
+
```

Comment:
I think this would cause duplicate `.o` files when linking. It may link but I think it may grow the resulting link. We don't `$(sort` the `OBJ` value to prevent this.

I tried a very simple example:
foo.c:
```c
int int1;
int int2;
```
main.c:
```c
extern int int1, int2;

int main() {
    return int1 + int2;
}
```
Linking `foo.o` once vs twice causes different-size executables:
```sh
$ gcc main.c foo.c
$ gcc -c main.c foo.c
$ gcc -o 1.out main.o foo.o
$ gcc -o 2.out main.o foo.o foo.o
$ ls -l ?.out
-rwxrwxr-x 1 halbert halbert 16552 Aug 11 13:36 1.out
-rwxrwxr-x 1 halbert halbert 16576 Aug 11 13:36 2.out
```
It's possible to do a boolean OR with gnu make, but it's not obvious:
https://stackoverflow.com/questions/7656425/makefile-ifeq-logical-or

We could `$(sort` to remove the duplicates. We already do that with the dirs:
https://github.com/adafruit/circuitpython/blob/1e53cf40ad5dba1c28f7f179c9f30b4f60ec3da2/py/mkrules.mk#L107-L112